types/wlr_layer_shell_v1.c

```diff
@@ -257,49 +256,45 @@ static void layer_surface_resource_destroy(struct wl_resource *resource) {
 	}
 }
 
-static bool layer_surface_state_changed(struct wlr_layer_surface_v1 *surface) {
-	struct wlr_layer_surface_v1_state *state;
+static bool layer_surface_state_changed(struct wlr_layer_surface_v1 *surface,
```

Comment:
I wonder if we shouldn't just get rid of this function and the `configure_next_serial` field in this patch already.

Reply:
Though actually, I don't think the layer shell protocol ever requires the compositor to send a noop configure, unlike the xdg-shell protocol. This is fine as-is I believe.

Reply:
On the other hand, if the compositor saves the configure serial returned and waits for the client to ack that serial, this function can cause problems. I'm back to thinking we should just get rid of it and behave in the most predictable way possible for compositors.

Reply:
Removed the function and the associated check, if the caller wants to send a configure then why not. `wlr_layer_surface_v1_configure()` now returns `surface->pending.configure_serial` on memory allocation failure, not like it matters much.